IdentityCore/src/cache/crypto/MSIDAssymetricKeyLookupAttributes.m

```diff
@@ -25,46 +25,43 @@
 
 @implementation MSIDAssymetricKeyLookupAttributes
 
+/*
+ This particular query generates the asymmetric key pair but only saves the private key in the keychain.
+ Please refer to https://developer.apple.com/documentation/security/certificate_key_and_trust_services/keys/generating_new_cryptographic_keys?language=objc
+ */
 - (NSDictionary *)assymetricKeyPairAttributes
 {
-    NSMutableDictionary *keyPairAttr = [NSMutableDictionary new];
-    keyPairAttr[(__bridge id)kSecAttrIsPermanent] = @YES;
-    keyPairAttr[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    keyPairAttr[(__bridge id)kSecAttrKeySizeInBits] = @2048;
-    keyPairAttr[(__bridge id)kSecAttrLabel] = self.keyDisplayableLabel;
-    keyPairAttr[(__bridge id)kSecAttrAccessible] = (__bridge id)kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly;
+    NSData *tag = [self.privateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary* attributes =
+        @{ (id)kSecAttrKeyType:               (id)kSecAttrKeyTypeRSA,
+           (id)kSecAttrKeySizeInBits:         @2048,
+           (id)kSecAttrLabel:self.keyDisplayableLabel,
```

Comment:
Previous code was using dictionary subscript which wouldn't crash if the value is nil.
New code will crash if e.g. self.keyDisplayableLabel is nil 
Can we revert?

Reply:
Addressed.